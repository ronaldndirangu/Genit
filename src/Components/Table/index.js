import React from "react";
import "./index.scss";

const Table = ({ numbers, handleSort }) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Index</th>
          <th className="table-header__no" onClick={e => handleSort(e)}>
            Number
          </th>
        </tr>
      </thead>
      <tbody>
        {numbers &&
          numbers.map((num, index) => {
            return (
              <tr key={index.toString()}>
                <td>{(index + 1).toString()}</td>
                <td>{num}</td>
              </tr>
            );
          })}
      </tbody>
    </table>
  );
};

export default Table;
