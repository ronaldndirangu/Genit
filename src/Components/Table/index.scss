table {
  font-family: monospace;
  font-size: 1.3rem;
  font-weight: 500;
  width: 60rem;
  box-shadow: 2px 4px 4px #ccc;
  margin: 1rem;
  background-color: #fff;
}

tbody {
  display: block;
  width: 100%;
  height: 40rem;
  overflow: auto;
}

td {
  padding: 0.8rem;
  text-align: center;
}

th {
  padding: 1rem;
  cursor: pointer;

  svg {
    margin: 0px 5px;
  }
}

table,
th,
td {
  border-bottom: 1px solid #ccc;
  border-collapse: collapse;
}
thead,
tbody tr {
  display: table;
  width: 100%;
  table-layout: fixed; /* even columns width , fix width of table too*/
}

th {
  color: #fff;
  text-transform: uppercase;
  background-color: #39298c99;
}

tr {
  cursor: default;
  transition: all 250ms ease-in-out 0s;
  &:hover {
    background-color: #39298c11;
  }
}
