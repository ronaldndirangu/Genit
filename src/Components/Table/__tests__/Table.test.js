import React from "react";
import { shallow } from "enzyme";
import Table from "../../Table";

const numbers = ["0722111333", "0719888444", "0703999888"];

describe("Table", () => {
  it("Table renders correctly", () => {
    const handleSort = jest.fn();
    const component = shallow(
      <Table numbers={numbers} handleSort={handleSort} />
    );
    component.find("th.table-header__no").simulate("click");
    expect(component).toMatchSnapshot();
    expect(handleSort).toHaveBeenCalled();
  });
});
