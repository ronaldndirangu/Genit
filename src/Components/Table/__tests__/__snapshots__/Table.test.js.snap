// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Table renders correctly 1`] = `ShallowWrapper {}`;
